### DISCUSSION TITLE
Why do proxy objects not have array_value() / object_value()?

### DISCUSSION THREAD

**Discussion by awolff64** [danielaparker/jsoncons#359] [Q&A] [answered]
I had the problem the when using the []-operator to access an object the returned value did not offer the array_value() / object_value() while it has al the as_...().

I am now using at() which return the a Json. However as I am converting existing json11 based code I have to touch many lines ...

Can anybody explain this?

**Comment by danielaparker** [comment, marked as answer]
I don't recall the `array_value()` and `object_value()` functions ever being part of the documented `basic_json` api, found [here](https://github.com/danielaparker/jsoncons/blob/master/doc/ref/basic_json.md). These are internal functions. The proxy should cover all the non-static functions that are defined in the documented api.

**Comment by awolff64** [comment]
Thanks for the quick reply.
I did not work against the documents api. Just used the auto completion of the development tool which offered me the ...value()-methods for basic_json-objects.
I understood that I should not use these methods at all. However from a software development point of view I liked them. They provide some type safety.
Thanks for all your work jsoncons is great!
